scripts/docker/pom-modify/PomFile.java

```diff
@@ -83,6 +86,35 @@ private void parseFields(String pom) {
             dBuilder = dbFactory.newDocumentBuilder();
             Document doc = dBuilder.parse(pomFile);
 
+	    // If pom file already patched, stop right here
+	    NodeList plugins = doc.getElementsByTagName("plugin");
```

Comment:
I don't think this will be a problem, but I just wanted to point out here that looking for the nodes with <plugin> could get nodes under other parents that we don't care about. If you look at the rewrite method, note that we have to parse down the structure (going project, then plugins) to get the specific structure we need to modify. Just looking for <plugin> nodes can get the <plugin> nodes under <pluginManagement><plugins>, which we do not want to add to.

However, since this logic is just checking that the testrunner plugin just exists after it has been added, either manually by the user or through this script, which either way would add it to the right place, then I think it is fine just to look at all <plugin> nodes regardless where they are, since it will eventually find it if it exists.

Reply:
Please read on a bit. I think the code below makes sure we have a build parent. Is that what you were referring to?

Reply:
Yes, that later code that checks the parent's parent is <build> is partially what I was referring to. A quick note on that is that it would technically match the case where the <build> is under another node, like <profile>, and the script does not add to that location. However, I think the logic of just looking across all <plugin> nodes for the testrunner one is still fine, as long as the user is adding by hand by the instructions or just using this script to add it.

Reply:
> it would technically match the case where the <build> is under another node, like <profile>, and the script does not add to that location

I see; then let me check if there's a nice way to get the full 'genealogy' of the plugin node to avoid ambiguities. I'll touch the branch again, anyway, for the warning.

Reply:
Done.